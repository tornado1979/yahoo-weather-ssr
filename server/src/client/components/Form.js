import React from 'react'
import propTypes from 'prop-types'

export const Form = ({ handleClick }) => {
  function submitForm(event) {
    event.preventDefault()
    const form = event.target
    const input = form.getElementsByTagName('input')[0]
    handleClick(input.value)
  }

  return (
    <div className="form-container">
      <form onSubmit={submitForm}>
        <input placeholder="type a city/zipcode" type="text" />
        <button className="btn" type="submit">Add city</button>
      </form>
    </div>
  )
}

Form.propTypes = {
  handleClick: propTypes.func.isRequired,
}
