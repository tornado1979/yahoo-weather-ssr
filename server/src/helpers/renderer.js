import React from 'react'
import { renderToString } from 'react-dom/server'
import { StaticRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { renderRoutes } from 'react-router-config'
import serialize from 'serialize-javascript'

import Routes from '../client/Routes'

export default (req, store, context) => {
  const content = renderToString(
    <Provider store={store}>
      <StaticRouter context={context} location={req.path}>
        <div>{renderRoutes(Routes)}</div>
      </StaticRouter>
    </Provider>,
  )
  return `
  <!DOCTYPE html>
  <html>
    <head>
      <meta charset="UTF-8">
      <title>Server Side render application</title>
    </head>
   <body>
     <div id="root">${content}</div>
     <script>
       window.INITIAL_STATE = ${serialize(store.getState())}
     </script>
     <script src="bundle.js"></script>
   </body>
  </html>
  `
}
