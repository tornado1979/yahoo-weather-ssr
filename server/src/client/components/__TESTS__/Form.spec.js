import React from 'react'
import renderer from 'react-test-renderer'

import { Form } from '../Form'

describe('Component Form', () => {
  it('renders correctly', () => {
    const handleClick = () => console.log('form submited')

    const tree = renderer
      .create(<Form handleClick={handleClick} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
