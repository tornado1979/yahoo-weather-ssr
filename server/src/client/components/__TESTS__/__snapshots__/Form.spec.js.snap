// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Form renders correctly 1`] = `
<div
  className="form-container"
>
  <form
    onSubmit={[Function]}
  >
    <input
      placeholder="type a city/zipcode"
      type="text"
    />
    <button
      className="btn"
      type="submit"
    >
      Add city
    </button>
  </form>
</div>
`;
